File: C++SrcFiles/LavaPunchScript.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cmath>
#include <algorithm>

using namespace std;

class LavaPunchScript : public MonoBehaviour
{
private:
    GameObject* player;
    float timer;
    float xpos;
    float ypos;

public:
    LavaPunchScript()
    {
        player = GameObject::FindGameObjectWithTag("Player");
        timer = 2.0f;
        xpos = transform->position.x;
        ypos = transform->position.y;

        if (transform->position.y <= -9.0f) {
            ypos = -8.0f;
            transform->position = Vector3(transform->position.x, ypos, transform->position.z);
        }

        if (transform->position.y <= -7.0f) {
            if (player->transform->position.x > transform->position.x) {
                xpos = player->transform->position.x;
            } else {
                xpos = player->transform->position.x;
            }
            transform->position = Vector3(xpos, transform->position.y, transform->position.z);
        }
    }

    void OnTriggerEnter2D(Collider2D* collision)
    {
        if (collision->gameObject->CompareTag("Player")) {
            cout << "Hit Target" << endl;
            collision->gameObject->GetComponent<SlimeScript>()->TakeDamage(20);
            cout << "Deleting bullet!" << endl;
        }
    }
};

File: C#SrcFiles/LavaPunchScript.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class LavaPunchScript : MonoBehaviour
{
    private GameObject player;
    //private float Distance;
   
    private float Xpos;
    private float Ypos;
    private float Timer;
    void Start()
    {
        player = GameObject.FindGameObjectWithTag("Player");
        Timer = 2f;
        Xpos = transform.position.x;
        Ypos = transform.position.y;
        if (transform.position.y <= -9f)
        {
            Ypos = -8f;
            transform.position = new Vector3(transform.position.x, Ypos, transform.position.z);
        }
        if (transform.position.y <= -7f)
        {
            if (player.transform.position.x > transform.position.x)
            {
                Xpos = player.transform.position.x;
            }
            else
            {
                Xpos = player.transform.position.x;
            }
            transform.position = new Vector3(Xpos, transform.position.y, transform.position.z);
        }
    }

    private void OnTriggerEnter2D(Collider2D collision)
    {
        if (collision.gameObject.CompareTag("Player"))
        {
            print("Hit Target");
            // giving 20 damage if bullet hits target;
            collision.gameObject.GetComponent<SlimeScript>().TakeDamage(20);
            print("Deleting bullet!");
        }
    }
}
